src/test/system_tests/sanity_build_test.js

```diff
@@ -4,114 +4,96 @@
-        console.error('Missing target_ip paramter!');
-        show_usage();
-        stop();
+    if (_.isUndefined(process.env.AWS_ACCESS) || _.isUndefined(process.env.AWS_SECRET)) {
+        console.error('Missing env credentials');
+        process.exit(5);
     }
 
-    await run_test(argv.target_ip, argv.upgrade_pack, false);
+    TEST_CTX.s3_endpoint = `http://${argv.s3_ip}:${argv.s3_port}`;
+    TEST_CTX.mgmt_endpoint = `ws://${argv.gmt_ip}:${argv.mgmt_port}`;
+
+    dbg.log0('Running Sanity Build Test');
+    await run_test();
     console.log('Finished Sanity Build Test Successfully');
     process.exit(0);
 }
 
-async function run_configuration_test(target_ip, mng_port) {
+async function run_test() {
+    await init_test();
+
     try {
-        await init_test(target_ip, mng_port);
         await create_configuration();
-    } catch (err) {
-        dbg.error('failed configuration test', err);
-        throw err;
+    } catch (error) {
+        console.warn('Configuration testing failure, caught', error);
+        process.exit(1);
     }
-}
-
-async function run_test(target_ip, upgrade_pack, dont_verify_version, skip_configuration) {
-    await init_test(target_ip);
 
     try {
-        if (!skip_configuration) {
-            await create_configuration();
-        }
+        await test_data();
     } catch (error) {
-        console.warn('Caught', error);
-        stop();
+        console.warn('Data testing failure, caught', error);
+        process.exit(1);
     }
-
-    await upgrade_and_test(target_ip, upgrade_pack, dont_verify_version);
-    await TEST_CTX.client.system.read_system();
 }
 
-async function init_test(target_ip, mng_port = '8080') {
+async function init_test() {
     const auth_params = {
         email: 'demo@noobaa.com',
         password: 'DeMo1',
         system: 'demo'
     };
     TEST_CTX._rpc = api.new_rpc();
     TEST_CTX.client = TEST_CTX._rpc.new_client({
-        address: 'ws://' + target_ip + ':' + mng_port
+        address: TEST_CTX.mgmt_endpoint
     });
     await TEST_CTX.client.create_auth_token(auth_params);
 
     TEST_CTX.bucketfunc = new BucketFunctions(TEST_CTX.client);
     TEST_CTX.cloudfunc = new CloudFunction(TEST_CTX.client);
 
-    TEST_CTX.aws_connection = TEST_CTX.cloudfunc.getAWSConnection();
+    TEST_CTX.aws_connection = TEST_CTX.cloudfunc.getAWSConnection(process.env.AWS_ACCESS, process.env.AWS_SECRET);
     TEST_CTX.aws_connection.name = `${TEST_CTX.aws_connection.name}12`;
-    TEST_CTX.azure_connection = blobops.AzureDefaultConnection;
+    TEST_CTX.azure_connection =
+        TEST_CTX.cloudfunc.getAzureConnection(process.env.AZURE_ID, process.env.AZURE_SECRET, process.env.AZURE_ENDPOINT);
     TEST_CTX.azure_connection.name = `${TEST_CTX.azure_connection.name}24`;
     TEST_CTX.compatible_v2 = {
         name: 'compatible_V2',
         endpoint_type: 'S3_COMPATIBLE',
-        endpoint: `http://${target_ip}`,
+        endpoint: `http://${TEST_CTX.s3_endpoint}`,
```

Comment:
this will result with:  http://http://${argv.s3_ip}:${argv.s3_port}
We should remove one http://